docker/api/image.py

```diff
@@ -81,10 +81,14 @@ def images(self, name=None, quiet=False, all=False, filters=None):
                 If the server returns an error.
         """
         params = {
-            'filter': name,
             'only_ids': 1 if quiet else 0,
             'all': 1 if all else 0,
         }
+        if utils.version_lt(self._version, 1.25):
```

Comment:
Could use some eyes on this one; my concern with this was; if no explicit version is set, will this pick this branch, or the other one (below)?

We should "assume" API v1.25 or up if no API version is set, so if the logic needs to be reversed for that, let me know

Reply:
With 'auto' as default, we now retrieve the engine's api version so `self._version` should be set to that or to one that was explicitly set via the version parameter. In case there is no version retrieved from the engine or none was set, I think we raise an error.

Reply:
Thanks! So current logic in this PR should probably good to go then

Reply:
LOL, except that it doesn't work if I see the CI output;

```
tests/integration/api_client_test.py:71: in test_resource_warnings
    client.images()
docker/api/image.py:87: in images
    if utils.version_lt(self._version, 1.25):
docker/utils/utils.py:70: in version_lt
    return compare_version(v1, v2) > 0
docker/utils/utils.py:60: in compare_version
    s2 = StrictVersion(v2)
/usr/local/lib/python2.7/distutils/version.py:40: in __init__
    self.parse(vstring)
/usr/local/lib/python2.7/distutils/version.py:105: in parse
    match = self.version_re.match(vstring)
E   TypeError: expected string or buffer
```

Reply:
Ah, need a string; pushed 🤞